### PR TITLE
Add handling of multiple front matter pages.

### PR DESCRIPTION
**Pull request by rvolz**
* The attribute "fm-files" can contain an array of filenames that will be inserted as EPUB front matter pages
* If the attribute is not set the package looks for a single "front-matter.html" file (current behavior)

### CODE REVIEW

**Comment by mojavelinux** [comment]
I like this idea of supporting more than one front matter file.

Could you file an issue before sending a PR? That way, we can discuss the design before proceeding with an implementation.

In that issue, I'd like to suggest that we auto-detect indexed file names (front-matter-1.html, front-matter-2.html, etc, sorted using the sort function). I don't like the idea of an attribute "fm-files" because that's not consistent with how we handle supplemental files elsewhere (and the name is too cryptic).

**Comment by rvolz** [comment]
Regarding the issues: no problem, I thought discussing code would be easier, but ok.

I'll change the PR to work with indexed file names. My preference was to name the pages according to their purpose (titlepage, epigraph etc.), but I can live with the other solution. 

However, one goal of the original solution was to keep the front matter files out of the project root directory, unlike `front-matter.html`. I would prefer to have at least a default or a configurable directory for them, like for the styles, converter destination or images.

**Comment by mojavelinux** [comment]
I know we tend to follow a different process than other projects, but I've found things move faster if we start with an issue, discuss ideas, then move to implementation. I will sometimes file and issue and a PR at the same time if it's already clear what needs to happen. But, regardless, the issue is important because it can be used to manage the code change and any related PRs. Personally, I don't think PRs should be treated like an issue since it mixes design and implementation.

> I'll change the PR to work with indexed file names. My preference was to name the pages according to their purpose (titlepage, epigraph etc.)

This is why we need a design phase ;)

> one goal of the original solution was to keep the front matter files out of the project root directory, unlike front-matter.html.

Right now the location is hardcoded, but it should perhaps be looking in something like frontmatterdir.

To be honest, the frontmatter feature was really just done for a very special use case, so it's not fully fleshed out yet.

**Comment by rvolz** [comment]
Ok, so the changes would be

* use indexed file names
* use an attribute like `epub3-frontmatterdir` (like with `epub3-stylesdir`)

Would that be ok, or would you need to think about that some more?

**Comment by mojavelinux** [comment]
I like the idea of epub3-frontmatterdir. That feels right. As for the files. I think we should just use a glob pattern to discover them, then sort them (perhaps just using a normal alpha sort). That way, it supports gaps and different naming schemes. The glob would be `front-matter*.html`, though perhaps that can be later overridden using epub3-frontmatterglob or something.

**Comment by rvolz** [comment]
Ok, that would works for me, I'll change the PR accordingly.

**Comment by mojavelinux** [comment]
Thanks! I've got the releases warmed up again, so I'll follow-up promptly this time.

File: lib/asciidoctor-epub3/packager.rb [hunk 1]
```diff
@@ -203,21 +203,50 @@ def add_cover_page doc, spine_builder, manifest
...
 
-  def add_images_from_front_matter
-    (::File.read 'front-matter.html').scan ImgSrcScanRx do
-      resources do
-        file $1
+  def get_frontmatter_files doc
+    if doc.attr? 'epub3-frontmatterdir'
+      fmdir = doc.attr 'epub3-frontmatterdir'
+      fmglob = 'front-matter.*\.html'
+      unless Dir.exist? fmdir
```
Comment (slonopotamus) [inline lib/asciidoctor-epub3/packager.rb]: `fmdir` needs to be resolved relative to `docdir` and not to current working directory. See #301

File: lib/asciidoctor-epub3/packager.rb [hunk 2]
```diff
@@ -203,21 +203,50 @@ def add_cover_page doc, spine_builder, manifest
...
       end
-    end if ::File.file? 'front-matter.html'
+      fms = Dir.entries(fmdir).delete_if {|x| !x.match fmglob}.sort.map {|y| File.join(fmdir,y)}
+      if fms && fms.length > 0
+        fms
+      else
+        warn %(Directory specified by 'epub3-frontmattderdir' contains no suitable files! Ignoring ...)
+        []
+      end
+    elsif File.exist? 'front-matter.html'
```
Comment (slonopotamus) [inline lib/asciidoctor-epub3/packager.rb]: Same problem, file is searched in process working directory.

File: lib/asciidoctor-epub3/packager.rb [hunk 3]
```diff
@@ -203,21 +203,50 @@ def add_cover_page doc, spine_builder, manifest
...
+    else
+      []
+    end
+  end
+
+  def add_images_from_front_matter doc
+    get_frontmatter_files(doc).each do |fmf|
+      (::File.read fmf).scan ImgSrcScanRx do
+        resources do
+          file $1
```
Comment (slonopotamus) [inline lib/asciidoctor-epub3/packager.rb]: Image file needs to be searched relative to front matter file.

**Comment by slonopotamus** [comment]
Thanks! Merged, with some corrections.

**Pull request merged**